Vostok.Logging.Abstractions/Extensions/FilterBySourceContextLogExtensions.cs

```diff
@@ -0,0 +1,65 @@
+using System.Linq;
+using JetBrains.Annotations;
+using Vostok.Logging.Abstractions.Values;
+
+namespace Vostok.Logging.Abstractions
+{
+     public static class FilterBySourceContextLogExtensions
+    {
+        public static ILog WithEventsSelectedBySourceContext(this ILog log, string context)
+        {
+            return new SourceContextFilterLog(log, context, true);
+        }
+
+        public static ILog WithEventsDroppedBySourceContext(this ILog log, string context)
+        {
+            return new SourceContextFilterLog(log, context, false);
+        }
+
+        public class SourceContextFilterLog : ILog
+        {
+            private readonly ILog baseLog;
+            private readonly string contextFilterValue;
+            private readonly bool filterAllowsEvent;
+
+            public SourceContextFilterLog([NotNull] ILog baseLog,
+                                          [CanBeNull] SourceContextValue sourceContextValue,
+                                          [NotNull] string contextFilterValue,
+                                          bool filterAllowsEvent)
+            {
+                this.baseLog = baseLog;
+                this.contextFilterValue = contextFilterValue;
+                this.filterAllowsEvent = filterAllowsEvent;
+                SourceContext = sourceContextValue;
+            }
+
+            public SourceContextFilterLog([NotNull] ILog baseLog,
+                                          [NotNull] string contextFilterValue,
+                                          bool filterAllowsEvent)
+                : this(baseLog, null, contextFilterValue, filterAllowsEvent)
+            {
+            }
+
+            [CanBeNull]
+            public SourceContextValue SourceContext { get; }
+
+            public void Log(LogEvent @event)
+            {
+                var matchFilter = SourceContext?.Contains(contextFilterValue) ?? false;
+                if (matchFilter == filterAllowsEvent)
+                {
+                    baseLog.Log(@event);
+                }
+            }
+
+            public bool IsEnabledFor(LogLevel level) => baseLog.IsEnabledFor(level);
+
+            public ILog ForContext(string context)
+            {
+                var baseLogForContext = baseLog.ForContext(context);
+                var newSourceContext = SourceContext + new SourceContextValue(context);
```

Comment:
No need to wrap into `SourceContextValue`, just add string.

Reply:
Also we can not to store SourceContext at all: our filter is always on of off for whole logger.
(Just disable or enable after ForContext(contextFilterValue) recieved.)

Reply:
> you can create an instance of SourceContextFilterLog with explicitly specified SourceContext
equvivalent to disable whole log?